File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int ans=-1;
        int i=0,j=height.length-1;
        while(i!=j){
            if(Math.min(height[i],height[j])*(j-i) > ans){
                ans = Math.min(height[i],height[j])*(j-i);
            }
            if (height[i]<height[j]){
                i++;
            }
            else
            {
                j--;
            }
        }
     return ans;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        max_water = 0
        left = 0
        right = len(height) - 1

        while left < right:
            h = min(height[left], height[right])
            width = right - left
            max_water = max(max_water, h * width)

            if height[left] < height[right]:
                left += 1
            else:
                right -= 1

        return max_water

        